lib/datadome.dart

```diff
@@ -106,11 +106,11 @@ class DataDome {
       'body': body
     };
 
-    final Map<dynamic, dynamic> response = await (_channel.invokeMapMethod('request', args) as FutureOr<Map<dynamic, dynamic>>);
-    Map<String, String> responseHeaders = new Map<String, String>.from(response['headers']);
+    final Map<dynamic, dynamic>? response = await (_channel.invokeMapMethod('request', args) as FutureOr<Map<dynamic, dynamic>?>);
+    Map<String, String> responseHeaders = new Map<String, String>.from(response?['headers']);
```

Comment:
I'm not sure `Map.from` accepts null argument in case `response` would be null. 
It would be better to handle the nullity and return an empty map instead. 

What is the rational behind always returning a non-null response even if all arguments are null?

Reply:
It doesn't seem that our native SDK handles the null status code. However, 

> "It would be better to handle the nullity and return an empty map instead."

 would not help as it's the response code and data that risks to be null, and the method expects to return an http Response not a map, which we cannot stub.

Reply:
There is a misunderstanding :) 
I don't ask for returning a map but I wanted to say `Map<String, String> responseHeaders = new Map<String, String>.from(response?['headers']);` can make the app crash if `response` is null, so if `response` is null, assign `{}` to `responseHeaders` .

Reply:
`It doesn't seem that our native SDK handles the null status code.` 
I don't really understand. The response is returned from native SDK, not given to native SDK, so whatever the native SDK handles no? 

But I notice that the return type of get, post, patch, delete ... is not nullable so may be that's why you need to always return non null response?